ntoskrnl/ps/job.c

```diff
@@ -86,52 +128,184 @@ PspInitializeJobStructures(VOID)
 NTSTATUS
 NTAPI
 PspAssignProcessToJob(PEPROCESS Process,
-    PEJOB Job)
+                      PEJOB Job)
 {
-    DPRINT("PspAssignProcessToJob() is unimplemented!\n");
-    return STATUS_NOT_IMPLEMENTED;
+    NTSTATUS Status = STATUS_SUCCESS;
+    DPRINT("PspAssignProcessToJob(%p,%p)\n", Process, Job);
+
+    /* Grab a reference for the lifetime of the process, released in ps/kill.c */
+    ObReferenceObject(Job);
+
+
```

Comment:
I wonder if it would be superfluous to assert that Process->Job is NULL here. When I started reading this code I was not sure what should happen in this case here but afterwards realised that the callers call it only when it is indeed NULL.

Reply:
I like it.